### DISCUSSION TITLE
Allow 'yield return' of enumerables somehow.

### DISCUSSION THREAD

**Discussion by michaelcheers** [dotnet/csharplang#5066] [Language Ideas]
In Javascript, they have `yield` which returns one value and `yield*` which returns multiple. I propose we should add something similar in c#. Eg. `yield return foreach` or `yield return*`.

Example:
```csharp
public static IEnumerable<T> Concat (IEnumerable<T> a, IEnumerable<T> b)
{
    yield return* a;
    yield return* b;
}
```

**Comment by HaloFour** [comment]
See: https://github.com/dotnet/csharplang/discussions/378

**Reply by michaelcheers** [reply]
It seems the conclusion of that was that it would not be done because of the performance implications.

**Reply by HaloFour** [reply]
I believe that's the case, or at least that a language feature wouldn't be considered without changes to iterators to make them performant in these cases.  I know that there have been prototypes but my understanding was that while they made nested iterables non-quadratic and perform much better that they performed worse on a simple iterator scenario, which is by far the more common case.
